test/common/sink_test.cc

```diff
@@ -0,0 +1,138 @@
+#include <filesystem>
+#include <fstream>
+
+#include "external/envoy/source/common/common/random_generator.h"
+
+#include "common/sink_impl.h"
+
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+namespace {
+
+// Future sink implementations register here for testing top level generic sink behavior.
+using SinkTypes = ::testing::Types<FileSinkImpl>;
+
+template <typename T> class TypedSinkTest : public ::testing::Test {
+public:
+  void SetUp() override { uuid_ = random_.uuid(); }
```

Comment:
Could we automatically clean up `"/tmp/nh/" + executionIdForTest() + "/"` in `TearDown`? (if it makes sense)

Reply:
Done in 94f7299. It's a little odd to do it here, as this may execute for other sink implementations when the TEST_P suites iterate over their parameterization, and will not write files to that location. But cleaning up is rather nice, and this won't cause trouble. 
So it seems pragmatic just add the two lines to clean up here.